var/spack/repos/builtin/packages/py-scikit-sparse/package.py

```diff
@@ -15,11 +15,16 @@ class PyScikitSparse(PythonPackage):
 
     maintainers("cgcgcg")
 
+    version("0.4.12", sha256="e6502fea9ba561cfa5491eb222ed2c81c16263d8182a293950db20509c941166")
+    version("0.4.11", sha256="64c61a8777b7c7ba8e1f2bf76bc767f740e6426f1cce2d90f1324b177618e1ca")
     version("0.4.8", sha256="2a224c60da3ef951975242ea777478583d3265efc72db5cfb7861686521a4009")
 
-    depends_on("python@3.6:3.11", type=("build", "link", "run"))
+    depends_on("python@3.6:3.10", type=("build", "link", "run"), when="@0.4.8")
+    depends_on("python@3.6:3.11", type=("build", "link", "run"), when="@0.4.11")
+    depends_on("python@3.6:3.11", type=("build", "link", "run"), when="@0.4.12")
     depends_on("py-setuptools@40.8:", type="build")
-    depends_on("py-cython@0.22:", type="build")
+    depends_on("py-cython@0.22:0.29", type="build", when="@:0.4.11")
+    depends_on("py-cython", type="build", when="@0.4.12")
```

Comment:
```suggestion
    depends_on("py-cython@0.22:", when="@0.4.12:", type="build")
```
see [pyproject.toml#L27](https://github.com/scikit-sparse/scikit-sparse/blob/v0.4.12/pyproject.toml#L27)

Reply:
Good catch!